eclass/tests/Makefile

```diff
@@ -0,0 +1,12 @@
+SH_FILES := $(wildcard *.sh)
+TEST_FILES := $(filter-out tests-common.sh, $(SH_FILES))
+TEST_OK_FILES := $(patsubst %.sh, .%.sh.ok,$ $(TEST_FILES))
+
+# A test is considered successful if the testfile itself did not change
+# (%.sh) and the contents of the eclass/ directory did not change (..).
+.%.sh.ok: %.sh ..
```

Comment:
I am actually unsure about the `..` part, as it does not detect a modified eclass. A simpler alternative, that does not cache the test results would probably be

```
.PHONY: test
test:
    find -type f -executable -exec {} \;
```